client/src/pages/Courses/Courses.js

```diff
@@ -17,11 +19,16 @@ const Courses = () => {
     const [isEditing, setIsEditing] = useState(false);
 
     const { data, columns, createCourse, updateCourse, deleteCourse, createCoursesFromUniApi, prefillCourse } = useCourses();
+    const history = useHistory();
+    const { courseId } = useParams();
 
     useEffect(() => {
         setSelectedCourse(isAddingCourse ? {} : data[currRow] || {});
         setIsEditing(false);
-    }, [data, currRow, isAddingCourse]);
+        const rowID = data.findIndex((course) => course._id === courseId);
+        setCurrRow(rowID.toString());
+        setIsAddingCourse(courseId === "new");
+    }, [data, currRow, isAddingCourse, courseId]);
```

Comment:
This useEffect seems to have many dependencies, could we split your `courseId` into a new useEffect to seperate concerns?